File: src/main/java/com/example/demo/mapper/CalendarHolidayMapper.java
package com.example.demo.mapper;

import com.example.demo.model.CalendarHoliday;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;

import java.time.LocalDate;
import java.util.List;

@Mapper
public interface CalendarHolidayMapper {
    List<CalendarHoliday> selectByYear(@Param("year") int year);

    List<CalendarHoliday> selectByDateRange(
            @Param("from") LocalDate from,
            @Param("to") LocalDate to
    );

    CalendarHoliday selectByDate(@Param("holidayDate") LocalDate holidayDate);

    int insert(CalendarHoliday holiday);

    int update(CalendarHoliday holiday);

    int deleteById(@Param("id") Long id);
}

File: src/main/kotlin/com/example/demo/mapper/CalendarHolidayMapper.kt
package com.example.demo.mapper

import com.example.demo.model.CalendarHoliday
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param
import java.time.LocalDate

@Mapper
interface CalendarHolidayMapper {
    fun selectByYear(@Param("year") year: Int): List<CalendarHoliday>

    fun selectByDateRange(
        @Param("from") from: LocalDate,
        @Param("to") to: LocalDate
    ): List<CalendarHoliday>

    fun selectByDate(@Param("holidayDate") holidayDate: LocalDate): CalendarHoliday?

    fun insert(holiday: CalendarHoliday): Int

    fun update(holiday: CalendarHoliday): Int

    fun deleteById(@Param("id") id: Long): Int
}

